x/dex/keeper/abci/end_block_liquidation.go

```diff
@@ -66,7 +66,8 @@ func (w KeeperWrapper) PlaceLiquidationOrders(ctx sdk.Context, contractAddr stri
 		pair := types.Pair{PriceDenom: order.PriceDenom, AssetDenom: order.AssetDenom}
 		orders := w.MemState.GetBlockOrders(typesutils.ContractAddress(contractAddr), typesutils.GetPairString(&pair))
 		order.Id = nextID
-		orders.AddOrder(order)
+		orderCopy := order
```

Comment:
why do we do a deep copy here

Reply:
yeah it's because we changed order cache from struct to a pointer to the struct in this PR